### PR TITLE
Profiles: Enable long press on duration field

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-3192

## What changed (plus any additional context for devs)

This PR enables long pressing on the year duration field on the confirm registration sheet.

## PoW (screenshots / screen recordings)

https://www.loom.com/share/730765f7982a4976b72915bb09ed3367

### CODE REVIEW

File: src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx [hunk 1]
```diff
@@ -11,17 +11,33 @@ import {
   Text,
 } from '@rainbow-me/design-system';
 import { REGISTRATION_MODES } from '@rainbow-me/helpers/ens';
+import { useInterval } from '@rainbow-me/hooks';
+
+const MIN_LONG_PRESS_DURATION = 200;
+const LONG_PRESS_INTERVAL = 100;
```
Comment (estebanmino) [inline src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx:17]: we have 69 in `FeesGweiInput` maybe we should use the same interval

File: src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx [hunk 2]
```diff
@@ -80,13 +120,9 @@ export default function RegistrationReviewRows({
...
                   <StepButton
-                    onPress={useCallback(
-                      () =>
-                        duration > 1
-                          ? onChangeDuration(duration - 1)
-                          : undefined,
-                      [duration, onChangeDuration]
-                    )}
+                    onLongPress={handleLongPressDecrement}
+                    onLongPressEnded={() => endLongPress()}
```
Comment (estebanmino) [inline src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx:124]:
```suggestion
                    onLongPressEnded={endLongPress}
```

File: src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx [hunk 3]
```diff
@@ -101,13 +137,9 @@ export default function RegistrationReviewRows({
...
                   <StepButton
-                    onPress={useCallback(
-                      () =>
-                        duration < maxDuration
-                          ? onChangeDuration(duration + 1)
-                          : undefined,
-                      [duration, maxDuration, onChangeDuration]
-                    )}
+                    onLongPress={handleLongPressIncrement}
+                    onLongPressEnded={() => endLongPress()}
```
Comment (estebanmino) [inline src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx:141]:
```suggestion
                    onLongPressEnded={endLongPress}
```

File: src/hooks/useInterval.ts [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
-  const handle = useRef();
+  const handle = useRef<number>();
 
   const start = useCallback((func, ms) => {
     handle.current = setInterval(func, ms);
   }, []);
 
   const stop = useCallback(
-    () => handle.current && clearInterval(handle.current),
+    () => (handle.current ? clearInterval(handle.current) : undefined),
```
Comment (estebanmino) [inline src/hooks/useInterval.ts:11]: out of curiosity, what was this change for?

Comment (jxom) [reply src/hooks/useInterval.ts:11]: TS yelled at me.

File: src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx (hunk 1 above)
Comment (jxom) [reply src/components/ens-registration/RegistrationReviewRows/RegistrationReviewRows.tsx:17]: sure

**Review by benisgold — APPROVED** [review]

**Pull request merged**